dotnet/targets/Xamarin.Shared.Sdk.targets

```diff
@@ -527,9 +527,13 @@
 			<!-- Yep, we want to run ILLink as well, because we need our custom steps to run (NativeAOT sets this to false, so set it back to true) -->
 			<RunILLink Condition="'$(PublishAot)' == 'true'">true</RunILLink>
 
-			<!-- Suppress trimmer warnings unless we're trimming all assemblies -->
-			<SuppressTrimAnalysisWarnings Condition="'$(SuppressTrimAnalysisWarnings)' == '' And '$(TrimMode)' == 'full'">false</SuppressTrimAnalysisWarnings>
-			<SuppressTrimAnalysisWarnings Condition="'$(SuppressTrimAnalysisWarnings)' == ''">true</SuppressTrimAnalysisWarnings>
+			<!--
+				In the case of NativeAOT builds, the trim warnings will be produced either by the ILLink roslyn analyzer in Debug builds
+				or by the ILC when publishing. We want to suppress the ILLink warnings in all build configurations to avoid duplicates.
+				We choose not to suppress the ILLink warnings when TrimMode is set to 'full' and we're not publishing. In this scenario
+				ILC will not run at a later stage of the build.
+			-->
+			<SuppressTrimAnalysisWarnings Condition="'$(PublishAot)' == 'true' And '$(_IsPublishing)' != 'true' And '$(TrimMode)' != 'full'">true</SuppressTrimAnalysisWarnings>
```

Comment:
@jonathanpeppers regarding your changes for Android:

We will need to add something like this in:  https://github.com/dotnet/android/blob/77f5fdd1d32b6ff2a2b36008f966cb4abffb8231/src/Xamarin.Android.Build.Tasks/Microsoft.Android.Sdk/targets/Microsoft.Android.Sdk.ILLink.targets#L12

and additionally the changes from lines 659-661:
https://github.com/xamarin/xamarin-macios/pull/20767/files#diff-2e24f1b6f44df7580244d00f106fa1a9e2257536fbca9d7bbdc74571549324a9R659-R661

the only difference would be that we would exchange `PublishAot` with `IsAotCompatible`